interface/wx/httprequest.h

```diff
@@ -0,0 +1,232 @@
+        @param value String value of the header
+
+        @since 3.1.2
+    */
+    void AddRequestHeader(const wxString& name, const wxString& value);
+
+    /**
+        Set HTTP method.
+
+        Set <a href="http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html">common</a>
+        or expanded HTTP method.
+
+        Overrides GET or POST methods that is used by default.
+
+        @param method
+            HTTP method name, e.g. "GET".
+
+        @since 3.1.2
+    */
+    void SetRequestMethod(const wxString& method);
+
+    /**
+        Set the text to be posted to the server.
+
+        After a successful call to this method, the request will use HTTP @c
+        POST instead of the default @c GET when it's executed.
+
+        @param data
+            The data to post.
+        @param contentType
+            The value of HTTP "Content-Type" header, e.g. "text/html;
+            charset=UTF-8".
+
+        @since 3.1.2
+    */
+    void SetRequestData(const wxString& data, const wxString& contentType);
+
+    /**
+        Set the binary data to be posted to the server.
+
+        The next request will
+        be an HTTP @c POST instead of the default HTTP @c GET and the given @a
+        data will be posted as the body of this request.
+
+        @since 3.1.2
+    */
+    void SetRequestData(const wxInputStream*, const wxString& contentType);
+
+    /**
+        Send the request to the server
+
+        @return
+            @true if no error occured or @false if a client or server side
+            error was triggered
+
+        @since 3.1.2
+    */
+    bool Send();
+
+    /**
+        Return a header from the response
+
+        @param name Name of the header field
+
+        @since 3.1.2
+    */
+    wxString GetResponseHeader(const wxString& name) const;
+
+    /**
+        Returns the status code returned by the server
+
+        @since 3.1.2
+    */
+    int GetResponseStatus() const;
+
+    /**
+        Returns the response status text
+
+        @since 3.1.2
+    */
+    wxString GetResponseStatusText() const;
+
+    /**
+        Returns a stream which represents the reponse data sent by the server.
+
+        @since 3.1.2
+    */
+    wxInputStream* GetResponseStream();
+
+    /**
+        Returns all response data as a string.
+
+        @since 3.1.2
+    */
+    wxString GetResponseData();
+};
+
+/**
+    @class wxHTTPSession
+
+    This class handles session wide parameters and data used by wxHTTPRequest
+    instances.
+
+    Every wxHTTPRequest sharing the same session object will use the same
+    cookies. Additionally an underlying network connection might be kept
+    alive to achive faster additional responses.
+
+    @since 3.1.2
+    @library{wxnet}
+    @category{net}
+
+    @see wxHTTPRequest
+*/
+class wxHTTPSession
+{
+public:
+    wxHTTPSession();
+
+    /**
+        Returns the default session used for all wxHTTPRequest objects
+
+        @since 3.1.2
+    */
+    static wxHTTPSession* defaultSession();
```

Comment:
Should be `GetDefaultSession()` in our naming convention. But IMO `wxHTTPSession::GetDefault()` is fine, no need to repeat "session".